lib/oban/queue/basic_engine.ex

```diff
@@ -130,6 +130,10 @@ defmodule Oban.Queue.BasicEngine do
   end
 
   @impl Engine
+  def complete_job(%Config{testing: :inline}, %Job{id: nil}) do
```

Comment:
I'm happy to take any direction on this change, as I feel that I'm potentially papering over what might be a valid/larger concern. I tried to limit the blast radius, though.

Reply:
The `:inline` option maps directly to the `InlineEngine`, so `BasicEngine` shouldn't ever be called with an inline config.